baybe/recommenders/base.py

```diff
@@ -16,20 +17,28 @@ class RecommenderProtocol(Protocol):
     def recommend(
         self,
         searchspace: SearchSpace,
+        objective: Optional[Objective],
```

Comment:
I'm not exactly sure if the is the perfect order of arguments or not, perhaps `objective` and `batch_size` should rather be swapped. Would be glad to hear your thoughts, but please consider the following:

Defining what is optional and what not turned out to be not as trivial as expected, and I'd ask you carefully check if my signature changes make sense to you everywhere. Note: they differ depending on the context, i.e.:
* sometimes stuff becomes optional at a lower level (e.g. the searchspace for our current meta recommenders, though not the meta recommender base class)
* and sometimes defaults are set / sometimes not, e.g. `objective` has a default in some base classes while the default disappears for `BayesianRecommender`. Note the difference between `Optional[Objective]` and `Optional[Objective]=None` though!!

Let me know

Reply:
they should definitely be swapped given that batch is also the required arg of the calling function and objective gets `added`

Reply:
Ok, should we then perhaps swap even further, like:
```python
    def recommend(
        self,
        batch_size: int,
        searchspace: SearchSpace,
        objective: Optional[Objective],
        train_df: Optional[pd.DataFrame],
    ) -> pd.DataFrame:
```

Reply:
But the above would also mean that `batch_size` can no longer have a default in concrete classes (which would be acceptable). So let me re-summarize: 

### Option 1 (without default for `batch_size`):
```python
    def recommend(
        self,
        batch_size: int,
        searchspace: SearchSpace,
        objective: Optional[Objective],
        measurements: Optional[pd.DataFrame],
    ) -> pd.DataFrame:
```
I like that `searchspace` and `objective` are next to each other

### Option 2 (with potential default for `batch_size`):
```python
    def recommend(
        self,
        searchspace: SearchSpace,
        batch_size: int,
        objective: Optional[Objective],
        measurements: Optional[pd.DataFrame],
    ) -> pd.DataFrame:
```
One could argue that having `batch_size=1` as default is reasonable

### Options 3 to N:
Any other permutation you consider reasonable

Reply:
I dont think `batch_size` needs to have a default enabled anywhere so im for option1

Reply:
done